### PR TITLE
Collaborative NTF Rework

### PR DESCRIPTION
**Pull request by 2cwldys**
## Corporate Nanotrasen Fleet Re-Work
Maintainers are allowed to assist with this project.

- - -
### (CARL, DO NOT ACCEPT THIS PULL REQUEST UNTIL WE ARE ABSOLUTELY READY!)
Planned Tweaks and Features:

- - [x] The NTSS Dagon can now be captained by the (NTF) Nanotrasen Fleet, instead of just the Exploratory Corps alone. [[1]](https://github.com/BoHBranch/BoH-Bay/pull/282/commits/b0401f2366446243955f29f7298bb5621248ccc6)

- - [ ] Flag Officers are now added to Bastion of Hestia.
- - [ ] Flag Officers outrank the commanding officer and executive officer and report /only/ to Central Command and in the certain of cases, relayed by the Corporate Executive.
    1) Flag Officers should be accompanied by Loss Prevention Associates over Corporate Executives.
    2) Flag Officers review the starship's operation and bridge crew.
    3) Flag Officers ensure orders and exploratory missions from Central Command are being followed and currently pursued.
    4) Flag Officers can take control of the vessel under the circumstances of an ineffective, or a traitorous captain.
    5) Flag Officers can take control of the vessel *at any time* by order of the Corporate Executive (authorized and faxed by administrative staff).
    6) Flag Officers can demote and promote (NTF) Nanotrasen Fleet crewmen with Corporate Executive counsel.
    7) Flag Officers are VIP protection targets and are usually targeted by traitors, round antagonists, or enemies of the corporation.
    8) Flag Officers can be used by administrative staff, or trusted individuals of the community to push more interesting rounds and conflicts.
    9) Flag Officers can make surprise visits mid-round per decision of administrative staff.
    10) Flag Officers cannot be chosen on round start without a slot being opened by administrative staff.
- - [x] Flag Officers will have their own quarters on the bridge deck. [[3]](https://github.com/BoHBranch/BoH-Bay/commit/26efb440593c0813a9b4805a4cda66d146b6c0c5#diff-57ae51efa245d2202ba0fabfa43c626c) [[4]](https://github.com/BoHBranch/BoH-Bay/pull/284)

- - [ ] A rank pip replicator vendor has been added.
    1) It can take rank pips found in the trash chute, lying around, or from demoted officers.
    2) It can produce rank pips to field promote individuals.
    3) The rank pip replicator can be found in the Deliberation Area, Corporate Executive Office, and the Commanding Officer's Quarters.
    4) It will have the ranks of enlisted crewmen (crewman apprentice to master chief petty officer), and commissioned officers (ensign to captain)
    5) Flag Officers, Commanding Officers, and the Corporate Executive can use the "Rank Pip Replicator" vendor.
- Abusing this mechanic can likely lead to administrative intervention, or discharge per the Corporate Executive; if you are not a traitor or antagonist in the round.

- - [x] (NTF) Nanotrasen Fleet Commissioned Officers are now available in the standard departments of the NTSS Dagon. (Engi, Med, Sci, Sec, Supply) [[2]](https://github.com/BoHBranch/BoH-Bay/pull/282/commits/726b50f0ad388835a6e180e393290baf01256bb2)
    1) Ensigns, Lieutenant Junior-Grades, and Lieutenants can now serve in common vessel departments, and dine in the Officer's Mess.
    2) They can serve as reserve bridge crew when the command staff are incapacitated, relocated, or in cryostasis by request of the chain of command.
    3) In order to pilot the helm, they must have a Pilot Qualification Pin.
    4) They can order (NTF) Nanotrasen Fleet enlisted non-commissioned officers, such as Petty Officers and Crewmen; and lastly, civilian contractors.
    5) Commissioned Officers are allowed on the bridge once allowed by their Commanding or Executive Officer, and may have access from the Emergency Armory on Red Alert.
    6) Commissioned Officers report to their department heads, per-usual; at the beginning of the round- and then their executive officer.
- Players should only consider commissioned officers if they are well versed in the diversity of their experience.
    (i.e a crewman apprentice should have less experience than a lieutenant)

### CODE REVIEW

File: maps/torch/job/command_jobs.dm [hunk 1]
```diff
@@ -155,7 +159,8 @@
 	allowed_ranks = list(
 		/datum/mil_rank/fleet/o3,
 		/datum/mil_rank/fleet/o4,
-		/datum/mil_rank/ec/o3
+		/datum/mil_rank/ec/o3,
+		/datum/mil_rak/ec/o4
```
Comment (NopemanMcHalt) [inline maps/torch/job/command_jobs.dm]: mistype oh god oh fuck

File: maps/torch/job/command_jobs.dm [hunk 2]
```diff
@@ -252,8 +258,9 @@
 	)
 	allowed_ranks = list(
 		/datum/mil_rank/ec/o3,
-		/datum/mil_rank/fleet/o2,
-		/datum/mil_rank/fleet/o3
+		/datum/mil_rank/ec/o4,
+		/datum/mil_rank/fleet/o3,
+		/datum/mil_rannk/fleet/o4,
```
Comment (NopemanMcHalt) [inline maps/torch/job/command_jobs.dm]: Same here

File: maps/torch/job/command_jobs.dm (hunk 1 above)
Comment (NopemanMcHalt) [inline maps/torch/job/command_jobs.dm]: Mistype here.

File: maps/torch/job/command_jobs.dm (hunk 2 above)
Comment (NopemanMcHalt) [inline maps/torch/job/command_jobs.dm]: Mistype here too.

File: maps/torch/job/command_jobs.dm (hunk 1 above)
Comment (2cwldys) [reply maps/torch/job/command_jobs.dm]: Thanks! I'll fix that right away!

File: maps/torch/job/command_jobs.dm (hunk 2 above)
Comment (2cwldys) [reply maps/torch/job/command_jobs.dm]: Oops! Will handle!

**Pull request closed without merging**